### PR TITLE
Exposed the EAGI FD as an io.Reader

### PR DESCRIPTION
**Pull request by angadn**
Would be handy to use this for Enhanced AGI - let me know if I've missed something on this one. 

_This may be considered as an interim push - and is **untested**. Reporting-in on real-world tests soon._

### CODE REVIEW

**Comment by Ulexus** [comment]
Thanks.

Rather than change the signature of the existing `New()`, can you just create a new `NewWithEAGI()`?    Better would be to adapt the system to use [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis), but that would mandate a bit more rearrangement, which I can do later.

**Comment by angadn** [comment]
Yes makes sense! Pushing the `NewWithEAGI` variant in a bit; will look at _functional options_ and see if I can help in the meanwhile.

File: agi.go [hunk 1]
```diff
@@ -138,7 +145,7 @@ func NewConn(conn net.Conn) *AGI {
 
 // NewStdio returns a new AGI session to stdin and stdout.
 func NewStdio() *AGI {
-	return New(os.Stdin, os.Stdout)
+	return NewWithEAGI(os.Stdin, os.NewFile(uintptr(3), "/dev/stdeagi"), os.Stdout)
```
Comment (Ulexus) [inline agi.go]: we cannot assume EAGI exists.  I'd make a new convenience wrapper function called NewEAGI() which does this.

Comment (angadn) [reply agi.go]: @Ulexus my bad - I assumed it wouldn't cause an error until Read was called. Have updated the PR.

**Comment by Ulexus** [comment]
Thanks, @angadn

**Pull request merged**